core/src/services/hdfs/backend.rs

```diff
@@ -95,6 +97,16 @@ impl HdfsBuilder {
         self.enable_append = enable_append;
         self
     }
+
+    /// Set temp dir for atomic write.
+    pub fn atomic_write_dir(&mut self, dir: &str) -> &mut Self {
```

Comment:
We can add a new test workflow in https://github.com/apache/incubator-opendal/blob/main/.github/workflows/service_test_hdfs.yml with `OPENDAL_HDFS_ATOMIC_WRITE_DIR=path/to/tmp` to test this PR.